docassemble/AssemblyLine/data/questions/ql_baseline.yml

```diff
@@ -418,54 +418,103 @@ comment: |
   That will be very context-specific.
 question: |
   % if al_form_type in ['starts_case','existing_case','appeal']:
-  How many children are involved in this case?
+  Are any children involved in this case?
   % else:
-  How many children are involved in this matter?
+  Are any children involved in this matter?
   % endif  
 fields:
-  - no label: children.target_number
+  - no label: children.there_are_any
+    datatype: yesnoradio
+  - How many?: children.target_number
     datatype: integer
+    show if: children.there_are_any
+validation code: |
+  if not children.there_are_any:
+    children.target_number = 0
+---
+# Customize this question if you know whose children they are in your interview!
+# this is just a very generic way to ask.
+id: children names without birthdate
+sets:
+  - children[0].name.first
+  - children[0].name.last
+question: |
+  % if hasattr(children, 'ask_number') and children.ask_number and children.target_number == 1:
+  Child's name
+  % else:
+  Name of first child
+  % endif
+fields:
+  - code: |
+      children[0].name_fields()
 ---
+# Customize this question if you know whose children they are in your interview!
```

Comment:
Does docassemble match the first question that will answer the field that it wants, or does it match the most specific? If it's the former, then the `children[i]` should be before the `children[0]` question.

Reply:
First it tries the most specific, and then furthest down in YAML file